src/openboardview/FileFormats/BRDFile.h

```diff
@@ -86,13 +87,20 @@ class BRDFile {
 	std::vector<BRDPin> pins;
 	std::vector<BRDNail> nails;
 
-	char *file_buf = nullptr;
-
 	bool valid = false;
 	std::string error_msg = "";
 
+	void AddNailsAsPins(const BRDPoint& max_point);
+	BRDFileBase() {}
+	~BRDFileBase() {}
+};
+
+class BRDFile : public BRDFileBase {
```

Comment:
Would it be possible to rename the current file to "BRDFileBase.h" (containing the parent `BRDFileBase` class, and the `READ_*` macros and shared `BRD*` structs for now) and extract the `BRDFile` class to a new "BRDFile.h" file?
`stringfile()` and `fix_to_utf8()` should also become part of a new "BRDFileBase.cpp" file, even though I'm not sure these functions are all that useful.

Reply:
Base class separated into its own file, includes also separated according to their real usage.